src/Npgsql/NpgsqlReadBuffer.Stream.cs

```diff
@@ -111,75 +111,94 @@ public override Task FlushAsync(CancellationToken cancellationToken)
                 => throw new NotSupportedException();
 
             public override int Read(byte[] buffer, int offset, int count)
-                => Read(buffer, offset, count, CancellationToken.None, false).GetAwaiter().GetResult();
+                => Read(new Span<byte>(buffer, offset, count));
```

Comment:
All the argument checks have disappeared (async as well)

Reply:
They are back.
I was relying on Span/Memory constructor to check them...